### PR TITLE
Integrate liquid staking v2 into vanilla runtime

### LINKED ISSUE #433: Finish stake client settlement
## Scope
- [x] finish stake client settlement

### LINKED ISSUE #435: Integrate staking pallet and client
## Motivation
Due to xcm::transact unstable, we still need stake-client as cross chain bridge.
## Scopes
- [x] Update record reward/slash.
- [x] Perform staking operations while receive `StakingOpRequest` event.
- [x] PROOF-OF-CONCEPT: call `settlement` periodically.
- [ ] Replace `setInterval` by listening `PeriodEnd` event if the design works.

### PR DESCRIPTION
**Pull request by alannotnerd**
Close #433 
Close #435

### CODE REVIEW

File: pallets/liquid-staking-v2/Cargo.toml [hunk 1]
```diff
@@ -8,20 +8,20 @@ version = '3.0.0'
...
 
 [dependencies]
-codec         = { package = 'parity-scale-codec', version = '2.2.0', features = ['max-encoded-len'], default-features = false }
-frame-support = { git = 'https://github.com/paritytech/substrate.git', branch = 'polkadot-v0.9.9', default-features = false }
-frame-system  = { git = 'https://github.com/paritytech/substrate.git', branch = 'polkadot-v0.9.9', default-features = false }
-orml-traits   = { git = 'https://github.com/open-web3-stack/open-runtime-module-library.git', default-features = false }
-sp-runtime    = { git = 'https://github.com/paritytech/substrate.git', branch = 'polkadot-v0.9.9', default-features = false }
-sp-std        = { git = 'https://github.com/paritytech/substrate.git', branch = 'polkadot-v0.9.9', default-features = false }
-xcm           = { git = 'https://github.com/paritytech/polkadot.git', branch = 'release-v0.9.9', default-features = false }
+codec           = { package = 'parity-scale-codec', version = '2.0.0', features= ["max-encoded-len"], default-features = false }
```
Comment (GopherJ) [inline pallets/liquid-staking-v2/Cargo.toml:11]: single quote here on `max-encoded-len`

File: pallets/liquid-staking-v2/Cargo.toml [hunk 2]
```diff
@@ -8,20 +8,20 @@ version = '3.0.0'
...
+frame-support   = { git = 'https://github.com/paritytech/substrate.git', branch = 'polkadot-v0.9.9', default-features = false }
+frame-system    = { git = 'https://github.com/paritytech/substrate.git', branch = 'polkadot-v0.9.9', default-features = false }
+orml-traits     = { git = 'https://github.com/open-web3-stack/open-runtime-module-library.git', default-features = false }
+sp-runtime      = { git = 'https://github.com/paritytech/substrate.git', branch = 'polkadot-v0.9.9', default-features = false }
+sp-std          = { git = 'https://github.com/paritytech/substrate.git', branch = 'polkadot-v0.9.9', default-features = false }
+xcm             = { git = 'https://github.com/paritytech/polkadot.git', branch = 'release-v0.9.9', default-features = false }
+orml-currencies = { git = 'https://github.com/open-web3-stack/open-runtime-module-library.git', default-features = false }
```
Comment (GopherJ) [inline pallets/liquid-staking-v2/Cargo.toml:18]: we dont need `orml-currencies` and `orml-tokens` IIRC, we just need `MultiCurrencyExtended` from orml-traits

File: pallets/liquid-staking-v2/Cargo.toml (hunk 1 above)
Comment (alannotnerd) [reply pallets/liquid-staking-v2/Cargo.toml:11]: okay, fix it in next commit

File: runtime/vanilla/Cargo.toml [hunk 3]
```diff
@@ -88,7 +88,7 @@ orml-xtokens                = { git = 'https://github.com/open-web3-stack/open-r
 
 # Parallel dependencies
 pallet-amm                   = { path = '../../pallets/amm', default-features = false }
-pallet-liquid-staking        = { path = '../../pallets/liquid-staking', default-features = false }
+pallet-liquid-staking        = { path = '../../pallets/liquid-staking-v2', package = "pallet-liquid-staking-v2", default-features = false }
```
Comment (GopherJ) [inline runtime/vanilla/Cargo.toml:91]: single quote here

File: runtime/vanilla/Cargo.toml [hunk 4]
```diff
@@ -111,12 +111,11 @@ runtime-benchmarks = [
   'frame-benchmarking',
   'frame-support/runtime-benchmarks',
   'frame-system-benchmarking',
-  'hex-literal',
```
Comment (GopherJ) [inline runtime/vanilla/Cargo.toml:114]: shall we remove this line? I'm not clear on this

File: runtime/vanilla/src/lib.rs [hunk 5]
```diff
@@ -431,26 +430,17 @@ parameter_types! {
...
     pub const StakingPalletId: PalletId = PalletId(*b"par/lqsk");
     pub const StakingCurrency: CurrencyId = CurrencyId::KSM;
     pub const LiquidCurrency: CurrencyId = CurrencyId::xKSM;
-    pub const MaxWithdrawAmount: Balance = 1_000_000_000_000_000;
-    pub const MaxAccountProcessingUnstake: u32 = 5;
+    pub RelayAgent: MultiLocation = MultiLocation::X2(
+        Junction::Parent,
+        Junction::AccountId32{
+            network: NetworkId::Any,
+            id: hex!["306721211d5404bd9da88e0204360a1a9ab8b87c66c1bc2fcdd37f3c2222cc20"]
```
Comment (GopherJ) [inline runtime/vanilla/src/lib.rs:441]: pls addd a comment on this account address

File: pallets/liquid-staking-v2/Cargo.toml (hunk 2 above)
Comment (alannotnerd) [reply pallets/liquid-staking-v2/Cargo.toml:18]: we need them due to trait requirement in runtime construction.

File: runtime/vanilla/src/lib.rs (hunk 5 above)
Comment (alannotnerd) [reply runtime/vanilla/src/lib.rs:441]: Okay

File: runtime/vanilla/Cargo.toml (hunk 4 above)
Comment (alannotnerd) [reply runtime/vanilla/Cargo.toml:114]: should be removed while it's not an optional requirement.

**Review by yz89 — APPROVED** [review]

**Comment by GopherJ** [comment]
let's wait CI

**Pull request merged**